dccrg.hpp

```diff
@@ -4386,6 +4386,14 @@ template <
 						{(uint64_t)x[0],(uint64_t)x[1],(uint64_t)x[2]},
 						std::max(refinement_level-1,0), 
 						std::min(refinement_level+1, this->mapping.get_maximum_refinement_level()));
+				// Check whether we are still inside the starting cell (which
```

Comment:
I would do:
```
uint64_t cellHere;
if (cell_at_location == starting_cell) {
	// Yup, we are.
	cellHere = starting_cell;
} else {
cellHere = this->get_existing_cell(
						{(uint64_t)x[0],(uint64_t)x[1],(uint64_t)x[2]},
						std::max(refinement_level-1,0), 
						std::min(refinement_level+1, this->mapping.get_maximum_refinement_level()));
}
```
so we only search for others if needed.

Reply:
Done!